src/Operations.jl

```diff
@@ -1993,6 +1999,22 @@ function status(env::EnvCache, registries::Vector{Registry.RegistryInstance}, pk
     if mode == PKGMODE_MANIFEST || mode == PKGMODE_COMBINED
         print_status(env, old_env, registries, header, filter_uuids, filter_names; diff, ignore_indent, io, outdated, silent_no_change)
     end
+    if is_manifest_current(env) === false
+        printpkgstyle(io, :Warning, """The project and manifest may be out of sync. \
+        The project hash recorded into the manifest when it was resolved does not match the hash of the current project. \
```

Comment:
This feels like an implementation detail of how the detection is made. No need to show that to a user?